### PR TITLE
Added SQL check server name availability operation

### PR DESCRIPTION
**Pull request by jaredmoo**
SDK test implemented at https://github.com/jaredmoo/azure-sdk-for-net/tree/checknameavailability

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by jaredmoo** [comment]
@nathannfan , @akromm , @jamestao 

Client code looks like this: https://github.com/jaredmoo/azure-sdk-for-net/blob/checknameavailability/src/SDKs/SqlManagement/Sql.Tests/CheckNameAvailabilityTests.cs#L22

**Comment by jaredmoo** [comment]
New linter warning: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/238462932#L1357
Won't fix because this is describing an existing API

New examples warnings: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/238462935#L1112
Won't fix, this is just more instances of the enum null issue

So I think there are no new warnings to fix.

**Comment by balajikris** [comment]
thanks @jaredmoo - I'm a little behind on getting to this, but rest assured. I'll take a look and help get this in, as soon as I can.

**Comment by balajikris** [comment]
_Notes:_

1. https://github.com/Azure/azure-rest-api-specs/pull/1215  will make a bunch of false linter errors with this composite spec go away.

**Comment by balajikris** [comment]
Now there are even more errors because multiple files were touched. Let me sort through the errors once - given the code added in the new commit, I really do not expect to find anything that we already don't know about.

**Comment by balajikris** [comment]
For reference, this PR touched only `sql.core.json` . We then decided to fix security definitions across all specs for this version of arm-sql, which made the PR touch all the files. The resultant failures are all already known and they belong to one of two classes - false errors or errors tracked by workitem #1128 which @jaredmoo is tracking and will be working on.

File: arm-sql/2014-04-01/swagger/sql.core.json [hunk 1]
```diff
@@ -16,7 +16,46 @@
...
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CheckNameAvailabilityRequest"
+            },
+            "description": "The required parameters for creating or updating a database."
```
Comment (balajikris) [inline arm-sql/2014-04-01/swagger/sql.core.json]: copy pasta?

**Review by balajikris — CHANGES_REQUESTED** [review]
Looks good to me, except for the one "description" which appears to be a copy paste error. Can you please fix that?

**Review by balajikris — APPROVED** [review]

**Pull request merged**